lib/rspec/rails/matchers/have_http_status.rb

```diff
@@ -255,12 +255,32 @@ def initialize(type)
             @invalid_response = nil
           end
 
-          # @return [Boolean] `true` if Rack's associated numeric HTTP code matched
-          #   the `response` code
+          # @api private
+          # Convert a symbol representing the expected response class into
+          #   the name of the test method to call on the response object
+          #
+          # @param response_symbol [Symbol] representing the expected http
+          #   response class
+          # @return [String] name of the response method, minus the ending
+          #   question mark
+          def response_method(response_symbol)
+            if 5 < ::Rails::VERSION::MAJOR ||
```

Comment:
We prefer conditionally defining methods rather than running `if` checks like this at runtime, after all it won't change for the duration of the spec suite.